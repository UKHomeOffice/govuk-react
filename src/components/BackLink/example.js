import React from "react";
import BackLink from ".";

export default () => <BackLink>Test</BackLink>;
