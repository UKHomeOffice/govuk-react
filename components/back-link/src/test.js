import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import sinon from 'sinon';

import BackLink from './';

describe('Back Link', () => {
  const onButtonClick = sinon.spy();
  const example = 'example';
  const wrapper = <BackLink goBack={onButtonClick}>{example}</BackLink>;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('simulates click events', () => {
    mount(wrapper)
      .find('button')
      .simulate('click');
    expect(onButtonClick).toHaveProperty('callCount', 1);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
