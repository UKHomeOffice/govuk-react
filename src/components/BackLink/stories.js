import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import BackLink from '.';

storiesOf('BackLink', module).add('Component default', () => (
  <BackLink goBack={action('go-back')}>Back</BackLink>
));
