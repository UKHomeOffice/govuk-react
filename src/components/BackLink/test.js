import React from "react";
import ReactDOM from "react-dom";
// import { shallow } from "enzyme";

import BackLink from "./";

describe("Back Link", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BackLink>Example</BackLink>, div);
  });
});
