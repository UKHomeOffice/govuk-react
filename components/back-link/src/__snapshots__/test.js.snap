// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back Link matches wrapper snapshot: wrapper mount 1`] = `
.emotion-1 {
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    margin-bottom: 15px;
  }
}

.emotion-0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571429;
  display: inline-block;
  position: relative;
  margin-top: 15px;
  padding-left: 14px;
  border: 0;
  color: #0b0c0c;
  border-bottom: 1px solid #0b0c0c;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 15px;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 16px;
    line-height: 1.25;
  }
}

.emotion-0::before {
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 50%,100% 100%,100% 0%);
  clip-path: polygon(0% 50%,100% 100%,100% 0%);
  border-width: 5px 6px 5px 0;
  border-right-color: #0b0c0c;
  content: '';
  position: absolute;
  top: -1px;
  bottom: 1px;
  left: 0;
  margin: auto;
}

.emotion-0:focus {
  background-color: #ffbf47;
  outline: 3px solid #ffbf47;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    margin-bottom: 15px;
  }
}

<Styled(BackLink)
  onClick={[Function]}
>
  <BackLink
    className="emotion-1"
    onClick={[Function]}
  >
    <Styled(button)
      className="emotion-1"
      onClick={[Function]}
    >
      <button
        className="emotion-0"
        onClick={[Function]}
      >
        example
      </button>
    </Styled(button)>
  </BackLink>
</Styled(BackLink)>
`;
